src/patternfly/layouts/Toolbar/styles.scss

```diff
@@ -2,18 +2,29 @@
 
 // The value of component scoped variables is always defined by a global variable:
 .pf-l-toolbar {
+  // Toolbar Layout
+  // adds negative margin to toolbar to offset the top margin given on all group divs
+  margin-top: calc(var(--pf-l-toolbar__group--MarginTop) * -1);
+
   // Toolbar Section
   --pf-l-toolbar__section--MarginTop: var(--pf-global--spacer--md);
 
   // Toolbar Group
+  --pf-l-toolbar__group--MarginTop: var(--pf-global--spacer--xl);
+  --pf-l-toolbar__group--MarginBottom: var(--pf-global--spacer--xl);
   --pf-l-toolbar__group--MarginRight: var(--pf-global--spacer--xl);
   --pf-l-toolbar__group--MarginLeft: var(--pf-global--spacer--xl);
 
   // Toolbar Item
+  --pf-l-toolbar__item--MarginTop: var(--pf-global--spacer--md);
   --pf-l-toolbar__item--MarginRight: var(--pf-global--spacer--md);
   --pf-l-toolbar__item--MarginLeft: var(--pf-global--spacer--md);
 }
 
+.pf-l-toolbar > .pf-l-toolbar__item {
```

Comment:
This only adds vertical space to items that are a direct descendent of toolbar. Looks like there is no margin between items when they are in a group.